client/src/components/ui/Icons.ts

```diff
@@ -0,0 +1,44 @@
+export interface Icon {
+    box: number; 
+    name: string;
+    svg: string;
+}
+
+export let iconsList: Array<Icon> = [
```

Comment:
We should consider using the [`svelte-preprocess-import-assets`](https://github.com/bluwy/svelte-preprocess-import-assets) plugin instead so that we can just write code that imports the SVG directly and inlines it into the bundle.

```html
<script>
    import Logo from './icons/logo.svg';
</script>
<img src={Logo} />
```

Looking at the `README` of the project, we simply have to install the dependency and add a few lines to the `svelte.config.js` file.

Reply:
It turns out that plugins are not necessary. See Parcel docs [on importing images and SVGs](https://parceljs.org/languages/svg/#url-references).

We can thus "component-ize" an SVG by following this scheme:

```html
<!-- client/src/Logo.svg -->
<script>
    // NOTE: We put the actual SVG in the `client/src/assets/` folder.
    const src = new URL('../assets/Logo.svg', import.meta.url);
</script>
<img {src} alt="Logo" />
```

Reply:
I added a template that others can use to generate their own icons. 
```typescript
<script lang="ts">
    import "./icon-styles.css"
    import { IconColor} from "../types.ts";
    
    export let name: string;
    export let color: IconColor = IconColor.Default;
    export let width: string = "1rem";
    export let height: string = "1rem";

    const iconUrl = new URL("../../assets/document-blank.svg", import.meta.url);

</script>

<img class="{color}" {width} {height} src="{iconUrl.pathname}" alt="{name}"/>

```
Overall, the only thing that should change between icons should be the string inside the `new Url`.